src/features/DisplayControls/Vitals/components/Vitals.jsx

```diff
@@ -0,0 +1,291 @@
+import React, { useEffect } from "react";
+import { Column, Row, Tile, SkeletonText } from "carbon-components-react";
+import { getPatientVitals } from "../utils/VitalsUtils";
+import { useState } from "react";
+import PropTypes from "prop-types";
+import "../styles/Vitals.scss";
+import { formatDate } from "../../../../utils/DateTimeUtils";
+import { formatDateAsString } from "../../../../utils/DateFormatter";
+import { FormattedMessage } from "react-intl";
+
+const Vitals = (props) => {
+  const { patientId } = props;
+  const [Vitals, setVitals] = useState({});
+  const [VitalUnits, setVitalUnits] = useState({});
+  const [VitalsDate, setVitalsDate] = useState(null);
+  const [VitalsTime, setVitalsTime] = useState(null);
+  const [isLoading, updateIsLoading] = useState(true);
+
+  const NoVitalsMessage = (
+    <FormattedMessage
+      id={"NO_VITALS_MESSAGE"}
+      defaultMessage={"No Vitals found for this patient"}
+    />
+  );
+
+  const vitalsHeaders = {
+    temperature: (
+      <FormattedMessage id={"Temperature"} defaultMessage={"Temp"} />
+    ),
+    bloodPressure: (
+      <FormattedMessage id={"BLOOD_PRESSURE"} defaultMessage={"BP"} />
+    ),
+    heartRate: (
+      <FormattedMessage id={"HEART_RATE"} defaultMessage={"Heart rate"} />
+    ),
+    respiratoryRate: (
+      <FormattedMessage id={"RESPIRATORY_RATE"} defaultMessage={"R.rate"} />
+    ),
+    weight: <FormattedMessage id={"WEIGHT"} defaultMessage={"Weight"} />,
+    height: <FormattedMessage id={"HEIGHT"} defaultMessage={"Height"} />,
+    spO2: <FormattedMessage id={"SPO2"} defaultMessage={"SpO2"} />,
+    BMI: <FormattedMessage id={"BMI"} defaultMessage={"BMI"} />,
+  };
+
+  var latestDate = null;
+
+  const setDateAndTime = (latestDateAndTime) => {
+    const dateAndTime = formatDate(latestDateAndTime).split(" ");
+    setVitalsDate(
+      formatDateAsString(new Date(latestDateAndTime), "DD/MM/YYYY")
+    );
+    setVitalsTime(dateAndTime.slice(3).join(" ").toUpperCase());
+  };
+
+  function getLatestDate(tabularData) {
+    for (const dateKey in tabularData) {
+      if (latestDate === null || dateKey > latestDate) {
+        latestDate = dateKey;
+      }
+    }
+  }
+
+  const mapVitalsData = (VitalsList) => {
```

Comment:
can we move this method out of this component and put it in vitals utils?

Reply:
If we move this method, I will have to make many changes in this file since it has dependencies